### PR TITLE
Fix decoration of setCaptureParams and recordDatastoreSegment

### PR DESCRIPTION
**Pull request by jderusse**

### CODE REVIEW

**Comment by jderusse** [comment]
Excelent suggestion @Nyholm I discovered 2 more typo.

I write a generic test based on reflection that asserts
- logger is called
- decorated is also called with the same argument that the decorator
- decorator returns the decorated response

File: Tests/NewRelic/LoggingInteractorDecoratorTest.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            ->willReturn($return);
+
+        $result = $interactor->$method(...$arguments);
+
+        $this->assertSame($return, $result);
+    }
+
+    public function provideMethods()
+    {
+        $reflection = new \ReflectionClass(NewRelicInteractorInterface::class);
```
Comment (Nyholm) [inline Tests/NewRelic/LoggingInteractorDecoratorTest.php]: Have you seen nyholm/nsa?

Comment (jderusse) [reply Tests/NewRelic/LoggingInteractorDecoratorTest.php]: I didn't know that package. The idea here is to retrieve methods with their arguments in order to automate calls, and it looks like this is not the purpose of this package. Did I missed a point?

Comment (Nyholm) [reply Tests/NewRelic/LoggingInteractorDecoratorTest.php]: No, I missed the point. (reviewed on my phone earlier) This peace of code you write is pretty cool. I may port it to nyholm/nsa.

**Pull request merged**